medical-portal/src/API/RSBC.DMF.MedicalPortal.API/Controllers/ChefsController.cs

```diff
@@ -140,6 +141,38 @@ public async Task<ActionResult> PutSubmission([FromQuery] string caseId, [FromBo
             }
             else if (status == SubmissionStatus.Final)
             {
+                Dictionary<string, object> chefsFlags = submission.Flags;
+
+                // get a list of all available Case Flags 
+                var f = cmsAdapterClient.GetAllFlags(new EmptyRequest());
+                if (f == null || f.Flags.Count == 0)
+                {
+                    logger.LogInformation("Could not find all flags in the CMS");
+                    return StatusCode((int)HttpStatusCode.NotFound,
+                        "Not found error - could not find all flags in the CMS");
+                }
+
+                IEnumerable<Flag> allCaseFlags = mapper.Map<IEnumerable<Flag>>(f.Flags);
+                
+                // if any Case Flags are present and active (true) in CHEFS, update Case and set IsCleanPass to false
+                var matchedFlags = allCaseFlags.Where(flag =>
+                    chefsFlags.ContainsKey(flag.FormId) && (bool)chefsFlags[flag.FormId]).ToArray();
+
+                UpdateCaseRequest updateCaseRequest = new UpdateCaseRequest()
+                {
+                    CaseId = caseId,
+                    IsCleanPass = matchedFlags.Length == 0
+                };
+
+                foreach (var item in matchedFlags)
```

Comment:
you can map the entire collection in one line, automapper will automap collections, even if they don't match e.g. IEnumerable to Repeated. If it's readonly, e.g. Repeated, you can use `AddRange(_mapper.Map...)`